httpx/middleware/query_normalizer.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"net/http"
+
+	"net/url"
+	"regexp"
+)
+
+// WithQueryNormalizer fixes wrong php query string handling as array
+func WithQueryNormalizer(h http.Handler) http.Handler {
+	rxp, e := regexp.Compile("\\[[0-9]+\\]")
+	if e != nil {
+		panic(e)
```

Comment:
Is panicing here a good idea?

Reply:
The only reason why there can be error is badly created regexp. If so, then why not panic?

Reply:
I suppose it is ok in this case, but rather use `MustCompile`.